### PR TITLE
Rope asr aed ngpt

### PR DESCRIPTION
**Pull request by nune-tadevosyan**
- Replace FlashAttention with torch based implementation
- Add masking for paddings

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/models/rnnt_bpe_models.py [hunk 1]
```diff
@@ -629,10 +629,9 @@ def optimizer_step(
         return ans
 
     def normalize_matrices(self):
-        with torch.no_grad():
```
Comment (pzelasko) [inline nemo/collections/asr/models/rnnt_bpe_models.py:632]: why remove torch.no_grad?

File: nemo/collections/asr/modules/ngpt_encoder.py [hunk 2]
```diff
@@ -16,9 +16,11 @@
 from collections import OrderedDict
 from dataclasses import dataclass
 
+from rotary_embedding_torch import RotaryEmbedding
```
Comment (pzelasko) [inline nemo/collections/asr/modules/ngpt_encoder.py]: double import

File: nemo/collections/asr/modules/ngpt_encoder.py [hunk 3]
```diff
@@ -208,36 +254,15 @@ def normalize_matrices(self):
...
-    position = torch.arange(n_positions, dtype=torch.float, device=device).unsqueeze(1)
-    div_term = torch.exp(torch.arange(0, dim, 2, dtype=torch.float, device=device) * (-math.log(10000.0) / dim))
-    sinusoidal_emb = torch.empty((n_positions, dim), device=device)
-    sinusoidal_emb[:, 0::2] = torch.sin(position * div_term)
-    sinusoidal_emb[:, 1::2] = torch.cos(position * div_term)
-    return sinusoidal_emb
-
-
-def justnorm(x, fp32: bool = False, idim: int = -1):
+def justnorm(x, fp32: bool = False, idim: int = -1,eps: float = 1e-10):
```
Comment (pzelasko) [inline nemo/collections/asr/modules/ngpt_encoder.py]: instead of 1e-10 which might be not representable in low precision types, use `torch.finfo(x.dtype).eps` https://docs.pytorch.org/docs/stable/type_info.html#torch.torch.finfo

File: nemo/collections/asr/models/rnnt_bpe_models.py (hunk 1 above)
Comment (nune-tadevosyan) [reply nemo/collections/asr/models/rnnt_bpe_models.py:632]: Missed it. Updated

File: nemo/collections/asr/modules/ngpt_encoder.py (hunk 2 above)
Comment (nune-tadevosyan) [reply nemo/collections/asr/modules/ngpt_encoder.py]: Removed

File: nemo/collections/asr/modules/ngpt_encoder.py (hunk 3 above)
Comment (nune-tadevosyan) [reply nemo/collections/asr/modules/ngpt_encoder.py]: Done
